ci/docker/benchmark_build.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+# Builds a docker image benchmark-dev:latest containing the benchmark scripts
+
+# Stop on errors.
+set -e
+
+# NOTE: explicit no -x for verbose commands. Because this is run in CI, doing so may result in
+# publishing sensitive information into public CI logs if someone makes a change in a
+# consuming script that is off guard.
+
+DOCKER_NAME="nighthawk-benchmark"
+DOCKER_IMAGE_PREFIX="envoyproxy/${DOCKER_NAME}"
+WORKSPACE="$(bazel info workspace)"
+BAZEL_BIN="$(bazel info -c opt bazel-bin)"
+TMP_DIR="${WORKSPACE}/tmp-docker-build-context"
+
+[ -d "${TMP_DIR}"  ] && rm -rf "${TMP_DIR}"
+
+echo "Preparing docker build context in ${TMP_DIR}"
+cp -r "${WORKSPACE}/ci/docker/" "${TMP_DIR}/"
+cp -rL "${BAZEL_BIN}/benchmarks" "${TMP_DIR}"
+
+cd "${TMP_DIR}"
+echo "running docker build ... "
+docker build -f "${TMP_DIR}/Dockerfile-${DOCKER_NAME}" -t "${DOCKER_IMAGE_PREFIX}-dev:latest" .
```

Comment:
I could very well be missing your intent here, but would it make sense for us to parameterize some of these arguments?  The one that seems to have the most utility to me is `latest` here.  Could we default to latest but let others specify other docker tags?

The other parameter I'm wondering about is TMP_DIR.  I've found in the past that requiring certain paths sometimes results in people being unable to proceed due to permissions errors on a directory.  Do you think that's a possible concern here, justifying a parameter?  Or is that just needless complexity

Reply:
We can certainly parameterize the image tag.

I can also use `mktemp` here for TMP_DIR.  I don't envision that this container will be built repeatedly, but yes, having a static name for a temporary path could present an issue.

I will update both of these.